spec/controllers/v3/branches_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V3::BranchesController do
+  subject { create :repository_compound }
```

Comment:
Actually, the subject of the test is the `BranchesController` instead of a repository, so please make this a `let!(:repository)` definition.